src/circuit/gadget.rs

```diff
@@ -35,45 +30,49 @@ impl super::Config {
         CommitIvkChip::construct(self.commit_ivk_config.clone())
     }
 
-    pub(super) fn ecc_chip(&self) -> EccChip<OrchardFixedBases> {
+    pub(super) fn ecc_chip(&self) -> EccChip<OrchardFixedBases, Lookup> {
         EccChip::construct(self.ecc_config.clone())
     }
 
     pub(super) fn sinsemilla_chip_1(
         &self,
-    ) -> SinsemillaChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases> {
+    ) -> SinsemillaChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases, Lookup> {
         SinsemillaChip::construct(self.sinsemilla_config_1.clone())
     }
 
     pub(super) fn sinsemilla_chip_2(
         &self,
-    ) -> SinsemillaChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases> {
+    ) -> SinsemillaChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases, Lookup> {
         SinsemillaChip::construct(self.sinsemilla_config_2.clone())
     }
 
     pub(super) fn merkle_chip_1(
         &self,
-    ) -> MerkleChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases> {
+    ) -> MerkleChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases, Lookup> {
         MerkleChip::construct(self.merkle_config_1.clone())
     }
 
     pub(super) fn merkle_chip_2(
         &self,
-    ) -> MerkleChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases> {
+    ) -> MerkleChip<OrchardHashDomains, OrchardCommitDomains, OrchardFixedBases, Lookup> {
         MerkleChip::construct(self.merkle_config_2.clone())
     }
 
     pub(super) fn poseidon_chip(&self) -> PoseidonChip<pallas::Base, 3, 2> {
         PoseidonChip::construct(self.poseidon_config.clone())
     }
 
-    pub(super) fn note_commit_chip_new(&self) -> NoteCommitChip {
+    pub(super) fn note_commit_chip_new(&self) -> NoteCommitChip<Lookup> {
         NoteCommitChip::construct(self.new_note_commit_config.clone())
     }
 
-    pub(super) fn note_commit_chip_old(&self) -> NoteCommitChip {
+    pub(super) fn note_commit_chip_old(&self) -> NoteCommitChip<Lookup> {
         NoteCommitChip::construct(self.old_note_commit_config.clone())
     }
+
+    pub(super) fn cond_swap_chip(&self) -> CondSwapChip<pallas::Base> {
+        CondSwapChip::construct(self.merkle_config_1.cond_swap_config.clone())
```

Comment:
In `halo2_gadgets 0.4.0` (specifically https://github.com/zcash/halo2/commit/a73ad8f80ae9b8a7c0967eac6223e920dd843581) I changed this to be a new method instead of partially-exposed fields. When `orchard 0.12.0` gets merged into the integration branch, this will become:
```suggestion
        CondSwapChip::construct(self.merkle_config_1.cond_swap_config().clone())
```